openml/extensions/sklearn/extension.py

```diff
@@ -0,0 +1,134 @@
+from typing import Any, Dict, List, Optional, Tuple, TYPE_CHECKING
+
+import sklearn.base
+
+from openml.extensions import Extension, register_extension
+from openml.extensions.sklearn.flow_functions import (
+    sklearn_to_flow,
+    flow_to_sklearn,
+    obtain_parameter_values,
+    get_version_information,
+    check_n_jobs,
+    is_estimator,
+    create_setup_string,
+    is_sklearn_flow,
+)
+from openml.extensions.sklearn.run_functions import (
+    seed_model,
+    run_model_on_fold,
+    is_hpo_class,
+    assert_is_hpo_class,
+    obtain_arff_trace,
+)
+
+
+if TYPE_CHECKING:
+    from openml.flows import OpenMLFlow
+    from openml.tasks.task import OpenMLTask
+    from openml.runs.trace import OpenMLRunTrace, OpenMLTraceIteration
+
+
+class SklearnExtension(Extension):
+
+    ################################################################################################
+    # General setup
+
+    @staticmethod
+    def can_handle_flow(flow: 'OpenMLFlow') -> bool:
+        return is_sklearn_flow(flow)
+
+    @staticmethod
+    def can_handle_model(model: Any) -> bool:
+        return isinstance(model, sklearn.base.BaseEstimator)
+
+    ################################################################################################
+    # Methods for flow serialization and de-serialization
+
+    def flow_to_model(self, flow: 'OpenMLFlow') -> Any:
+        return flow_to_sklearn(flow)
+
+    def model_to_flow(self, model: Any) -> 'OpenMLFlow':
+        return sklearn_to_flow(model)
+
+    def flow_to_parameters(self, flow: Any) -> List:
+        return obtain_parameter_values(flow)
+
+    def get_version_information(self) -> List[str]:
+        return get_version_information()
+
+    def create_setup_string(self, model: Any) -> str:
```

Comment:
I don't understand what this function does. `setup_string` as in the legacy OpenML setup string? This can potentially be removed.

Reply:
`setup_string` as it can be uploaded for a run. Is this legacy? If it is, I think removing it from the Python connector should be part of a different pull request.

Reply:
Agreed.